src/bin/repl/helper.rs

```diff
@@ -0,0 +1,113 @@
+use super::PREFIX;
+use ansi_term::Style;
+use rustyline::{
+    completion::{extract_word, Completer},
+    highlight::{Highlighter, MatchingBracketHighlighter},
+    hint::Hinter,
+    line_buffer::LineBuffer,
+    validate::{ValidationContext, ValidationResult, Validator},
+    Context,
+};
+use rustyline_derive::Helper;
+use std::borrow::Cow;
+
+#[derive(Helper)]
+pub struct ReplHelper {
+    highlighter: MatchingBracketHighlighter,
+    commands: Vec<&'static str>,
+}
+
+impl ReplHelper {
+    pub fn new(commands: Vec<&'static str>) -> Self {
+        Self {
+            commands,
+            highlighter: Default::default(),
+        }
+    }
+}
+
+impl Highlighter for ReplHelper {
+    fn highlight<'l>(&self, line: &'l str, pos: usize) -> Cow<'l, str> {
+        // TODO: Syntax highlighting.
+        self.highlighter.highlight(line, pos)
+    }
+
+    fn highlight_hint<'h>(&self, hint: &'h str) -> Cow<'h, str> {
+        let hint = Style::new().dimmed().paint(hint);
+        Cow::Owned(hint.to_string())
```

Comment:
This could be `Cow::Borrowed` I think.
```suggestion
        Cow::Borrowed(hint)
```

Reply:
Unfortunately, it can't.

Reply:
Hmm I see, and you can't change the return type since it's a trait. Ok, not worth worrying about then.